addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanner.java

```diff
@@ -124,6 +125,20 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         };
     }
 
+    private static Predicate<Element> htmlNotInline() {
+        // 'else' branch won't ever execute since everything evaluated is a SCRIPT/LINK.
+        // Leaving it in place for clarity.
+        return element -> {
+            if (element.getName() == HTMLElementName.SCRIPT) {
+                return !element.getAttributeValue("src").startsWith("data:");
```

Comment:
The test fails on this line with a `NullPointerException`. Not sure why, since the test HTML message includes a "src" attribute inside the script. What am I missing?

Reply:
Oh, the failing test wasn't the one I wrote but `shouldNotRaiseAlertGivenElementIsInline`.

Reply:
I'll add null checks in both places.

Reply:
This (method) is duplicating what `SupportedElements` does/provides, the scheme check should be added to `#getHost(...)`.

Reply:
On it, thanks.